iXBRLViewerPlugin/viewer/src/js/search.js

```diff
@@ -53,20 +60,25 @@ ReportSearch.prototype.buildSearchIndex = function () {
     })
 }
 
-ReportSearch.prototype.search = function(s) {
-    var rr = this._searchIndex.search(s);
+ReportSearch.prototype.search = function (s) {
+    var rr = this._searchIndex.search(s.searchString);
     var results = []
     var searchIndex = this;
 
-    if (s == "") {
-        return [];
-    }
-
-    rr.forEach((r,i) => 
-        results.push({
-            "fact": searchIndex._report.getItemById(r.ref),
-            "score": r.score
-        })
+    rr.forEach((r,i) => {
+            var item = searchIndex._report.getItemById(r.ref);
+            if (
+                ((!item.isHidden() && s.showVisibleFacts) || (item.isHidden() && s.showHiddenFacts)) &&
```

Comment:
`item.isHidden() ? s.showHiddenFacts : s.showVisibleFacts` might be clearer.

Similarly `s.conceptTypeFilter == '*' || s.conceptTypeFilter == (item.isNumeric() ? 'numeric' : 'text')`.

Reply:
Yep, that's much better.  Done in b6805b3